### PR TITLE
normalize error interfaces

### LINKED ISSUE #61: Events should emit token IDs
https://github.com/valorem-labs-inc/valorem-core/pull/59 - let's discuss the feedback from @thal0x on emitting token IDs in error reversions here.

### PR DESCRIPTION
**Pull request by 0xAlcibiades**
Closes #61

### CODE REVIEW

File: src/interfaces/IOptionSettlementEngine.sol [hunk 1]
```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
...
 
     /// @notice This claimId has already been claimed.
-    error AlreadyClaimed();
+    /// @param claimId Supplied claim ID.
+    error AlreadyClaimed(uint256 claimId);
 
     /// @notice You can't claim before expiry.
-    error ClaimTooSoon();
+    /// @param claimId Supplied claim ID.
+    error ClaimTooSoon(uint256 claimId);
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:80]: Would it be useful to emit the expiry timestamp here in addition to the claimID? That way, the error emitted would have information necessary for caller to handle.

File: src/interfaces/IOptionSettlementEngine.sol [hunk 2]
```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
     error ExpiredOption(uint256 optionId, uint40 expiry);
 
     /// @notice This option cannot yet be exercised.
-    error ExerciseTooEarly();
+    /// @param optionId Supplied option ID.
+    error ExerciseTooEarly(uint256 optionId);
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:64]: should we emit the exercise timestamp as well, for the benefit of the caller

File: src/interfaces/IOptionSettlementEngine.sol (hunk 1 above)
Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol:80]: a guiding principle we might consider for errors is to always provide enough information in the error to make it actionable. @0xAlcibiades

File: src/interfaces/IOptionSettlementEngine.sol [hunk 3]
```diff
@@ -27,7 +27,8 @@ interface IOptionSettlementEngine {
     error OptionsChainExists(bytes32 hash);
 
     /// @notice The expiry timestamp is less than 24 hours from now.
-    error ExpiryTooSoon();
+    /// @param optionId Supplied option ID.
+    error ExpiryTooSoon(uint256 optionId);
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:31]: Supply expiry timestamp

File: src/interfaces/IOptionSettlementEngine.sol [hunk 4]
```diff
@@ -27,7 +27,8 @@ interface IOptionSettlementEngine {
     error OptionsChainExists(bytes32 hash);
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:27]: this error name should change also

File: src/interfaces/IOptionSettlementEngine.sol [hunk 5]
```diff
@@ -27,7 +27,8 @@ interface IOptionSettlementEngine {
...
 
     /// @notice The expiry timestamp is less than 24 hours from now.
-    error ExpiryTooSoon();
+    /// @param optionId Supplied option ID.
+    error ExpiryTooSoon(uint256 optionId);
 
     /// @notice The option exercise window is less than 24 hours long.
     error ExerciseWindowTooShort();
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:34]: emit supplied exercise window

File: src/interfaces/IOptionSettlementEngine.sol (hunk 3 above)
Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol:31]: nit: standardize on
/**
*/
for multi line comments

File: src/interfaces/IOptionSettlementEngine.sol [hunk 6]
```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
     error ExpiredOption(uint256 optionId, uint40 expiry);
 
     /// @notice This option cannot yet be exercised.
-    error ExerciseTooEarly();
+    /// @param optionId Supplied option ID.
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:63]: nit: standardize on
/**
*/
for multi line comments

File: src/interfaces/IOptionSettlementEngine.sol [hunk 7]
```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
...
+    error ExerciseTooEarly(uint256 optionId);
 
     /// @notice This option has no claims written against it.
-    error NoClaims();
+    /// @param optionId Supplied option ID.
+    error NoClaims(uint256 optionId);
 
     /// @notice This account has no claims.
-    error BalanceTooLow();
+    /// @param claimId Supplied claim ID.
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:71]: nit: standardize on
/**
*/
for multi line comments

File: src/interfaces/IOptionSettlementEngine.sol [hunk 8]
```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
...
+    error BalanceTooLow(uint256 claimId);
 
     /// @notice This claimId has already been claimed.
-    error AlreadyClaimed();
+    /// @param claimId Supplied claim ID.
+    error AlreadyClaimed(uint256 claimId);
 
     /// @notice You can't claim before expiry.
-    error ClaimTooSoon();
+    /// @param claimId Supplied claim ID.
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:79]: nit: standardize on
/**
*/
for multi line comments

File: src/interfaces/IOptionSettlementEngine.sol [hunk 9]
```diff
@@ -59,19 +60,24 @@ interface IOptionSettlementEngine {
...
 
     /// @notice This option has no claims written against it.
-    error NoClaims();
+    /// @param optionId Supplied option ID.
+    error NoClaims(uint256 optionId);
 
     /// @notice This account has no claims.
-    error BalanceTooLow();
+    /// @param claimId Supplied claim ID.
+    error BalanceTooLow(uint256 claimId);
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:72]: I understand this error is likely to align on the ERC1155 impl, but the message seems slightly opaque. We could consider renaming this error as e.g. "**`RedeemerDoesNotOwnClaimId`**", and perhaps emit the owner of the claim ID supplied.

File: src/interfaces/IOptionSettlementEngine.sol (hunk 4 above)
Comment (0xAlcibiades) [reply src/interfaces/IOptionSettlementEngine.sol:27]: To what?

Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol:27]: sry, `OptionsTypeExists`.

Comment (0xAlcibiades) [reply src/interfaces/IOptionSettlementEngine.sol]: Merged in master, which had that already.

File: src/interfaces/IOptionSettlementEngine.sol [hunk 10]
```diff
@@ -26,8 +26,11 @@ interface IOptionSettlementEngine {
      */
     error OptionsTypeExists(uint256 optionId);
 
-    /// @notice The expiry timestamp is less than 24 hours from now.
-    error ExpiryTooSoon();
+    /**
+     * @notice The expiry timestamp is less than 24 hours from now.
+     * @param optionId Supplied option ID.
```
Comment (Flip-Liquid) [inline src/interfaces/IOptionSettlementEngine.sol:31]: @param for expiry

**Review by Flip-Liquid — APPROVED** [review]
changes look good

**Pull request merged**